packages/@adobe/spectrum-css-temp/components/table/index.css

```diff
@@ -284,6 +284,10 @@ governing permissions and limitations under the License.
   padding-top: 0px;
   padding-bottom: 0px;
   vertical-align: var(--spectrum-table-cell-checkbox-vertical-alignment);
+
+  .spectrum-Table-checkbox {
+    padding-inline-end: 0px;
```

Comment:
The fix works without this, but it feels more correct to explicitly call out that we don't want this padding.